### PR TITLE
fixes #53: added std AnyVal/AnyRef typeclass instances

### LINKED ISSUE #53: add typeclass instances for AnyVal
we should have Show instances in std for things like Float, Double, Int, Long, etc

### PR DESCRIPTION
**Pull request by mandubian**

### CODE REVIEW

File: std/src/main/scala/cats/std/anyval.scala [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  extends IntInstances
+  with    ByteInstances
+  with    CharInstances
+  with    LongInstances
+  with    ShortInstances
+  with    FloatInstances
+  with    DoubleInstances
+  with    BooleanInstances
+  with    UnitInstances
+  with    BigIntInstances
```
Comment (kionka) [inline std/src/main/scala/cats/std/anyval.scala]: ```BigInt``` is an ```AnyRef```, but I could see how you might want to include it. One could probably make a similar argument for ```BigDecimal```. http://www.scala-lang.org/api/current/index.html#scala.math.BigInt

Comment (mandubian) [inline std/src/main/scala/cats/std/anyval.scala]: Oups, I think I was a bit too mechanical :)
Let's change it & add BigDecimal

**Comment by stew** [comment]
:+1:

**Comment by stew** [comment]
one minor point? Do we need anyval? if we do, is it to be similar to scalaz? and if so, do we want to make that anyVal to actually match scalaz?

I think my vote would be to just lose anyval. I don't feel strongly about any of the three alternatives, however.

**Comment by non** [comment]
Yeah I would be happy to just have `cats.std.<concrete-type>` and then `cats.std.all` personally.

**Comment by non** [comment]
(otoh I'm happy to merge this now and then cross that bridge later :+1: )

**Comment by mandubian** [comment]
I hesitated about anyval & anyref & I don't care if we remove them.
Generally people need a special type or all but not anyval.

**Comment by stew** [comment]
@mandubian lets kill them and merge then

**Comment by stew** [comment]
thanks @mandubian !

**Pull request merged**